src/classes/BDI_MigrationMappingUtility.cls

```diff
@@ -0,0 +1,388 @@
+            BDI_HelpTextCMT.MappingDetail mappingDetail = helpTextFieldMap.get(key);
+            setImportStatusAndLookupFields(mappingDetail, importStatusAndLookupMap);
+        }
+
+        return helpTextFieldMap;
+    }
+
+    /**
+     * @description Set import lookup and import status fields on the MappingDetail instance
+     * @param  mappingDetail Instance of MappingDetail where fields are to be set
+     * @param  importStatusAndLookupMap Map holding the field values to be set to MappingDetail
+     */
+    private void setImportStatusAndLookupFields(BDI_HelpTextCMT.MappingDetail mappingDetail, 
+        Map<String, BDI_HelpTextCMT.ImportStatusAndLookupParts> importStatusAndLookupMap) {
+
+        BDI_HelpTextCMT.ImportStatusAndLookupParts importStatusAndLookupParts = 
+                importStatusAndLookupMap.get(mappingDetail.dataImportObjectName);
+
+        mappingDetail.importedRecordField = importStatusAndLookupParts.importLookupFieldName;
+        mappingDetail.importedRecordStatusField = importStatusAndLookupParts.importStatusFieldName;
+    }
+
+    /**
+     * @description Builds an instance of ImportStatusAndLookupParts which holds an object name, 
+     *              ImportedLookup__c and ImportStatus__c field names
+     * @param dataImportFieldName Name of a Data Import field
+     * @param importStatusAndLookupMap List of ImportStatusAndLookupParts instances
+     * @return instance of ImportStatusAndLookupParts
+     */
+    private BDI_HelpTextCMT.ImportStatusAndLookupParts getImportStatusAndLookupParts(String dataImportFieldName, 
+            Map<String, BDI_HelpTextCMT.ImportStatusAndLookupParts> importStatusAndLookupMap) {
+
+        // Create new instance of ImportStatusAndLookupParts
+        BDI_HelpTextCMT.ImportStatusAndLookupParts importRelatedField = 
+            new BDI_HelpTextCMT.ImportStatusAndLookupParts(dataImportFieldName);
+
+        // Check if there is already an existing and matching instance of ImportStatusAndLookupParts
+        BDI_HelpTextCMT.ImportStatusAndLookupParts exists = 
+            importStatusAndLookupMap.get(importRelatedField.objectName);
+
+        // If one exists, reset instance with newly added field
+        if (exists != null) {
+            importRelatedField = new BDI_HelpTextCMT.ImportStatusAndLookupParts(exists, dataImportFieldName);
+        }
+
+        return importRelatedField;
+    }
+
+    /**
+     * @description Queries for default Object Mappings and builds a map of DeveloperName to Object Mapping record
+     * Used to check for existance of Object Mapping record without loop
+     * @param objectMappingSetId Id of the default Object Mapping Set
+     * @return Collection of Object Mapping DeveloperName to Object Mapping record
+     */
+    private Map<String, Data_Import_Object_Mapping__mdt> getDefaultObjectMappings(Id objectMappingSetId) {
+        List<Data_Import_Object_Mapping__mdt> existingObjectMappings = [SELECT 
```

Comment:
@psadie similar here, do we want to do any handling of the case where the object mapping may not exist?